src/VisualStudio/CSharp/Impl/ProjectSystemShim/CSharpProjectShim.IServiceProvider.cs

```diff
@@ -10,8 +12,25 @@ internal partial class CSharpProjectShim : Microsoft.VisualStudio.OLE.Interop.IS
     {
         public int QueryService(ref Guid guidService, ref Guid riid, out IntPtr ppvObject)
         {
-            var serviceProvider = (Microsoft.VisualStudio.OLE.Interop.IServiceProvider)ServiceProvider;
-            return serviceProvider.QueryService(ref guidService, ref riid, out ppvObject);
+            if (riid == typeof(IVsContainedLanguageFactory).GUID)
```

Comment:
❕ This needs a comment and a link to the bug which, once fixed, means we can remove this workaround.